lib/chains/data_extraction_chain.ex

```diff
@@ -88,7 +89,7 @@ defmodule LangChain.Chains.DataExtractionChain do
       |> LLMChain.add_messages(messages)
       |> LLMChain.run()
       |> case do
-        {:ok, _updated_chain, %Message{role: :function_call, arguments: %{"info" => info}}}
+        {:ok, _updated_chain, %Message{role: :function, arguments: %{"info" => info}}}
```

Comment:
after specifying possible values for message `:role` dialyzer caught this 🙌🏻 
https://github.com/brainlid/langchain/blob/95217c19ea81515b119355f25be66e6b5c2f3bb6/lib/message.ex#L45-L47

Reply:
now it's fixed by https://github.com/brainlid/langchain/pull/16
but still I think the project would benefit from Dialyzer 🙂